### ISSUE TITLE
Package build string harmonization with tensorflow and pytorch

### ISSUE THREAD

**Issue by hmaarrfk** [conda-forge/onnxruntime-feedstock#83]
### Comment:

Would it acceptable to harmonize the build string with that of tensorflow and pytorch

onnxruntime (this feedstock)

```
{% set build_ext = "cuda" if cuda_enabled else "cpu" %}
  string: py{{ CONDA_PY }}h{{ PKG_HASH }}_{{ PKG_BUILDNUM }}_{{ build_ext }}
```

The main features that are missing:
1. The version of cuda that things are compiled with is not immediately obvious.
2. `cuda` comes at the end, whereas it arrives at the beginning in tensorflow and pytorch

Tensorflow:
https://github.com/conda-forge/tensorflow-feedstock/blob/main/recipe/meta.yaml#L136
```

      string: cuda{{ cuda_compiler_version | replace('.', '') }}py{{ CONDA_PY }}h{{ PKG_HASH }}_{{ PKG_BUILDNUM }}  # [cuda_compiler_version != "None"]
      string: cpu_py{{ CONDA_PY }}h{{ PKG_HASH }}_{{ PKG_BUILDNUM }}                                                # [cuda_compiler_version == "None"]
```

Pytorch
https://github.com/conda-forge/pytorch-cpu-feedstock/blob/main/recipe/meta.yaml
```
      string: cuda{{ cuda_compiler_version | replace('.', '') }}py{{ CONDA_PY }}h{{ PKG_HASH }}_{{ PKG_BUILDNUM }}  # [cuda_compiler_version != "None"]
      string: cpu_{{ blas_impl }}_py{{ CONDA_PY }}h{{ PKG_HASH }}_{{ PKG_BUILDNUM }}                                                # [cuda_compiler_version == "None"]
```

**Comment by traversaro** [comment]
The initial choice for the build string currently used is detailed in https://github.com/conda-forge/onnxruntime-feedstock/pull/63#issuecomment-1614740936 . In a nutshell, I tried to take inspiration from existing packages, but personally I do not have any strong opinion on this.

**Comment by traversaro** [comment]
> The initial choice for the build string currently used is detailed in [#63 (comment)](https://github.com/conda-forge/onnxruntime-feedstock/pull/63#issuecomment-1614740936) . In a nutshell, I tried to take inspiration from existing packages, but personally I do not have any strong opinion on this.

Ok, looking a bit it seems that I copied the one used in libarrow: https://github.com/conda-forge/arrow-cpp-feedstock/blob/cb3550db9a03b0a24d71427dcf0a21939d74fe15/recipe/meta.yaml#L112C19-L112C27 / pyarrow: https://github.com/conda-forge/arrow-cpp-feedstock/blob/cb3550db9a03b0a24d71427dcf0a21939d74fe15/recipe/meta.yaml#L624 , for me it is fine to change.

**Comment by timkpaine** [comment]
yes these were opened at the same time: https://github.com/conda-forge/pyarrow-feedstock/issues/149

**Comment by hmaarrfk** [comment]
To note, the answer was the same in both issues raised:
https://github.com/conda-forge/onnxruntime-feedstock/issues/144#issuecomment-2869786500
